nsqd/http.go

```diff
@@ -494,7 +494,24 @@ func (s *httpServer) doStats(w http.ResponseWriter, req *http.Request, ps httpro
 	channelName, _ := reqParams.Get("channel")
 	jsonFormat := formatString == "json"
 
-	stats := s.ctx.nsqd.GetStats()
+	var (
+		state TopicStats
+		stats []TopicStats
+	)
+	if topicName == "" {
+		stats = s.ctx.nsqd.GetStats()
+	} else if channelName == "" {
+		state, err = s.ctx.nsqd.GetTopicStats(topicName)
+	} else {
+		state, err = s.ctx.nsqd.GetChannelStats(topicName, channelName)
+	}
+	if nil != err {
```

Comment:
minor: we typically prefer the variable in question to be on the _left_ side of the comparison, e.g. `if err != nil {`